pkg/backup/compaction_dist_test.go

```diff
@@ -0,0 +1,209 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+package backup
+
+import (
+	"context"
+	"testing"
+
+	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/jobs/jobspb"
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/security/username"
+	"github.com/cockroachdb/cockroach/pkg/sql/execinfrapb"
+	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
+	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCreateCompactionCorePlacements(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	type Key = roachpb.Key
+	testcases := []struct {
+		name                 string
+		entries              []execinfrapb.RestoreSpanEntry
+		numNodes             int
+		expectedDistribution [][]roachpb.Span
+	}{
+		{
+			name: "single node, single span entry",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
+			},
+			numNodes: 1,
+			expectedDistribution: [][]roachpb.Span{
+				{{Key: Key("a"), EndKey: Key("b")}},
+			},
+		},
+		{
+			name: "single node, multiple span entries",
+			entries: []execinfrapb.RestoreSpanEntry{
+				{Span: roachpb.Span{Key: Key("a"), EndKey: Key("b")}},
```

Comment:
uber nit: to reduce the chars of code you could use some closures similar to this [test](https://github.com/kev-cao/cockroach/blob/backup/compaction-planning/pkg/crosscluster/physical/stream_ingestion_dist_test.go#L240): so the test case could like: `entries: entries(sp(a,b),sp(c,d)...`

Reply:
Updated! Looks a lot more readable now for sure